util/src/main/java/com/google/cloud/hadoop/util/testing/MockHttpTransportHelper.java

```diff
@@ -107,35 +104,6 @@ public String getErrorDomain() {
 
   private MockHttpTransportHelper() {}
 
-  public static HttpResponse fakeResponse(String header, Object headerValue, InputStream content)
-      throws IOException {
-    return fakeResponse(ImmutableMap.of(header, headerValue), content);
-  }
-
-  public static HttpResponse fakeResponse(Map<String, Object> headers, InputStream content)
-      throws IOException {
-    HttpTransport transport =
-        new MockHttpTransport() {
-          @Override
-          public LowLevelHttpRequest buildRequest(String method, String url) {
-            return new MockLowLevelHttpRequest() {
-              @Override
-              public LowLevelHttpResponse execute() {
-                MockLowLevelHttpResponse response = new MockLowLevelHttpResponse();
-                headers.forEach((h, hv) -> response.addHeader(h, String.valueOf(hv)));
-                return response.setContent(content);
-              }
-            };
-          }
-        };
-    HttpRequest request =
-        transport
-            .createRequestFactory()
-            .buildGetRequest(HttpTesting.SIMPLE_GENERIC_URL)
-            .setResponseReturnRawInputStream(true);
-    return request.execute();
-  }
-
```

Comment:
Hm, I think that it maybe better to not delete this code if it could be used for other tests? Or it's useful only for Mockito tests?

Reply:
I think the new added arbitrary behavior simulation method can achieve the same thing. This basically obtain response at runtime from the underlying input stream to allow inject custom input steam behavior (like throwing exception, returning arbitrary content from input stream, etc).